R/sensitivity_labels.R

```diff
@@ -143,19 +153,102 @@ apply_sensitivity_label <- function(file, label) {
     )
   }
 
-  # Check file is Excel workbook
-  file_extension <- tolower(tools::file_ext(file))
-  if (!file_extension %in% c("xlsx", "xls")) {
-    cli::cli_abort(
-      "{.arg file} must be an Excel workbook with {.val .xlsx} or {.val .xls} extension, not {.val .{file_extension}}."
-    )
+  # Check file is valid
+  file_ext <- tolower(tools::file_ext(file))
+  if (!file_ext %in% c("xlsx", "xls", "docx")) {
+    cli::cli_abort("{.arg file} must be an Excel workbook or Word document with{.val .xlsx}, {.val .xls}, or {.val .docx} extension, not {.val .{file_ext}}.")
   }
 
-  # Load workbook and apply label
-  wb <- openxlsx2::wb_load(file)
-  xml <- sensitivity_label_xml[[label]]
-  wb <- openxlsx2::wb_add_mips(wb, xml)
-  openxlsx2::wb_save(wb, file)
+  ## Apply label to Excel workbooks ----
+
+  if(file_ext %in% c("xlsx", "xls")){
+    wb <- openxlsx2::wb_load(file)
+    xml_map <- .get_sensitivity_xml_map()
+    xml <- xml_map[[label]]
+    wb <- openxlsx2::wb_add_mips(wb, xml)
+    openxlsx2::wb_save(wb, file)
+  }
+
+  ## Apply label to Word docs ----
+
+  if(file_ext == "docx"){
```

Comment:
does '.doc` file type need included for old compatibility or are they structurally different from `docx` so the same approach might not work? If complicated to investigate, don't worry about it.

Reply:
that's great thanks, no need to add in the `.doc` file type in that case